users.js

```diff
@@ -95,6 +96,56 @@ var lockedUsers = Object.create(null);
 var lockedRanges = Users.lockedRanges = Object.create(null);
 var rangelockedUsers = Object.create(null);
 
+function importLocksBans() {
+	fs.readFile('config/locksbans.csv', function (err, data) {
+		if (err) return;
+		var now = Date.now();
+		data = ('' + data).split("\n");
+		for (var i = 0; i < data.length; i++) {
+			if (!data[i]) continue;
+			var row = data[i].split(",");
+			var time = parseInt(row[1]);
+			if (now - time > PERSISTENT_PUNISHMENT_TIME) continue;
+			var entry = {
+				time: time,
+				userid: row[2]
+			};
+			if (row[0] === 'banip') {
+				bannedIps[row[3]] = entry;
+			} else if (row[0] === 'ban') {
+				bannedUsers[row[3]] = entry;
+			} else if (row[0] === 'lockip') {
+				lockedIps[row[3]] = entry;
+			} else if (row[0] === 'lock') {
+				lockedUsers[row[3]] = entry;
+			}
+		}
+	});
+}
+function exportLocksBans() {
+	var buffer = '';
+	for (var ip in bannedIps) {
+		if (!bannedIps[ip].time) continue;
+		buffer += "banip," + bannedIps[ip].time + "," + bannedIps[ip].userid + "," + ip + "\n";
```

Comment:
Apparently we _don't_ periodically clean them up.

We should.

You know how when importing, we don't import bans/locks that have already expired?

We shouldn't export bans/locks that have already expired. In fact, we should delete them while exporting.